src/main/java/cz/tstrecha/timetracker/service/impl/UserServiceImpl.java

```diff
@@ -140,22 +140,29 @@ public RelationshipDTO updateRelationship(RelationshipCreateUpdateRequestDTO req
 
     @Override
     @Transactional
-    public String hasPermissionToChangeContext(Long id, UserContext userContext) {
+    public LoginResponseDTO hasPermissionToChangeContext(Long id, UserContext userContext) {
         var contextUserDTO = userContext.getRelationshipsReceiving().stream()
                 .filter(relation -> relation.getId().equals(id))
                 .findFirst()
                 .orElseThrow(() -> new PermissionException("User dont have permission to change context to id [" + id + "]"));
 
         var userEntity = userRepository.findById(userContext.getId()).orElseThrow();
 
-        return JwtAuthenticationFilter.AUTHORIZATION_HEADER_BEARER_PREFIX + authenticationService.generateToken(userEntity, contextUserDTO);
+        var token = authenticationService.generateToken(userEntity, contextUserDTO);
+        var refreshToken = authenticationService.generateRefreshToken(userEntity.getId(), contextUserDTO.getId());
```

Comment:
This generates token for logged user and authorizes it as logged user, you shoud call @InjectLoggedUser.id